### PR TITLE
Error and crash reporter

### LINKED ISSUE #15: Display errors in edit window
It's a little annoying to wait for hover tooltip on parse errors in the fileList view. The error message should probably show up in the edit menu when double clicking, or maybe double clicking on that cell will just pop up a small error message.

### PR DESCRIPTION
**Pull request by Schultzer**
Resolves #15 #14
- [x] Add send mechanism
- [x] Add optional email field for non-anonymized
- [x] build function to send error reports with data anonymously
- [x] make it look fabulous with some CSS styles 
- [x] Make crash reporter work
- [x] Add specs for error report
- ~~Add specs for crash report~~
- [x] Set up server to receive reports

### CODE REVIEW

File: app/report.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+<head>
+  <meta charset="utf-8">
+  <title>Blazing Bookkeeper</title>
+
+  <link rel="stylesheet" type="text/css" href="assets/css/app.css">
+  <link rel="stylesheet" type="text/css" href="assets/css/report.css">
+
+  <script src="report.js"></script>
+</head>
+<body id="main" v-on:keydown.prevent.27="close" v-on:keydown.prevent.13="send">
```
Comment (danschultzer) [inline app/report.html]: It shouldn't send on `enter`.

File: src/background.js [hunk 2]
```diff
@@ -7,12 +7,16 @@ import { app, Menu, ipcMain, shell, BrowserWindow } from 'electron';
 import { devMenuTemplate } from './menu/dev_menu_template';
 import { editMenuTemplate } from './menu/edit_menu_template';
 import createWindow from './helpers/window';
+import thirdparty_env from './utils/thirdparty_env'
```
Comment (danschultzer) [inline src/background.js]: Why is this here?

Comment (danschultzer) [inline src/background.js]: Also L16-17

File: src/crash-reporter-start.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+module.exports = function (extra) {
```
Comment (danschultzer) [inline src/crash-reporter-start.js]: This should be tested.

File: src/report.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import { remote, webFrame, ipcRenderer, nativeImage } from 'electron';
```
Comment (danschultzer) [inline src/report.js]: Why is `nativeImage` loaded here?

File: src/file_list/file_list.spec.js [hunk 5]
```diff
@@ -129,8 +129,7 @@ describe("fileList", function() {
             date: "2016-05-05",
             amount: "6,000.00"
           };
-
-          callback(null, parsed);
+          callback(JSON, parsed);
```
Comment (danschultzer) [inline src/file_list/file_list.spec.js]: What is this for?

File: src/file_list/file_list.js [hunk 6]
```diff
@@ -98,6 +98,9 @@ export class FileList {
         var tmpFile = tmp.tmpNameSync({ postfix: name.substr(name.lastIndexOf(".")) }),
           stream = jetpack.createReadStream(path),
           finishedCallback = (error, result) => {
+            if (error) {
+              error.json = JSON.parse(JSON.stringify(error, Object.getOwnPropertyNames(error)));
```
Comment (danschultzer) [inline src/file_list/file_list.js]: This should probably be moved out to passing the variable to other windows.

**Review by danschultzer — APPROVED** [review]

**Comment by Schultzer** [comment]
👍 LGTM
